src/types/Types.hs

```diff
@@ -809,6 +883,19 @@ isRefType ty
 isUnresolved Type{inner = Unresolved{}} = True
 isUnresolved _ = False
 
+getAdtTag Type{inner = ClassType{adtTag}} = adtTag
+getAdtTag _ = 0
```

Comment:
If this function is only supposed to be called on class types, it's better to throw an error here!